app/services/update_query_from_params_service.rb

```diff
@@ -77,11 +79,19 @@ def apply_columns(params)
   end
 
   def apply_sums(params)
-    query.display_sums = params[:display_sums] if params[:display_sums]
+    query.display_sums = params[:display_sums] if params.key?(:display_sums)
   end
 
   def apply_timeline(params)
-    query.timeline_visible = params[:timeline_visible] if params[:timeline_visible]
+    query.timeline_visible = params[:timeline_visible] if params.key?(:timeline_visible)
+  end
+
+  def apply_hierarchy(params)
+    if query.grouped?
+      query.show_hierarchies = false
+    else
+      query.show_hierarchies = params[:show_hierarchies] if params.key?(:show_hierarchies)
+    end
```

Comment:
I don't think we'll need this restriction anymore and instead, let the user receive the full error message.